Applications/spire/source/charting/chart_view.cpp

```diff
@@ -248,6 +264,19 @@ void ChartView::showEvent(QShowEvent* event) {
   QWidget::showEvent(event);
 }
 
+void ChartView::update_auto_scale() {
+  if(m_candlesticks.size() > 0) {
+    auto auto_scale_top = ChartValue(-INT_MAX);
```

Comment:
```-INT_MAX``` is always a bug, period. At any rate candlesticks almost certainly do not represent ```int```s so using an extreme int value will result in bugs. Find another solution to this problem.